homeassistant/components/nws/const.py

```diff
@@ -101,82 +98,93 @@
 OBSERVATION_VALID_TIME = timedelta(minutes=20)
 FORECAST_VALID_TIME = timedelta(minutes=45)
 
-SENSOR_TYPES = {
-    "dewpoint": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Dew Point",
-        ATTR_UNIT: TEMP_CELSIUS,
-        ATTR_UNIT_CONVERT: TEMP_CELSIUS,
-    },
-    "temperature": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Temperature",
-        ATTR_UNIT: TEMP_CELSIUS,
-        ATTR_UNIT_CONVERT: TEMP_CELSIUS,
-    },
-    "windChill": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Wind Chill",
-        ATTR_UNIT: TEMP_CELSIUS,
-        ATTR_UNIT_CONVERT: TEMP_CELSIUS,
-    },
-    "heatIndex": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Heat Index",
-        ATTR_UNIT: TEMP_CELSIUS,
-        ATTR_UNIT_CONVERT: TEMP_CELSIUS,
-    },
-    "relativeHumidity": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Relative Humidity",
-        ATTR_UNIT: PERCENTAGE,
-        ATTR_UNIT_CONVERT: PERCENTAGE,
-    },
-    "windSpeed": {
-        ATTR_DEVICE_CLASS: None,
-        ATTR_ICON: "mdi:weather-windy",
-        ATTR_LABEL: "Wind Speed",
-        ATTR_UNIT: SPEED_KILOMETERS_PER_HOUR,
-        ATTR_UNIT_CONVERT: SPEED_MILES_PER_HOUR,
-    },
-    "windGust": {
-        ATTR_DEVICE_CLASS: None,
-        ATTR_ICON: "mdi:weather-windy",
-        ATTR_LABEL: "Wind Gust",
-        ATTR_UNIT: SPEED_KILOMETERS_PER_HOUR,
-        ATTR_UNIT_CONVERT: SPEED_MILES_PER_HOUR,
-    },
-    "windDirection": {
-        ATTR_DEVICE_CLASS: None,
-        ATTR_ICON: "mdi:compass-rose",
-        ATTR_LABEL: "Wind Direction",
-        ATTR_UNIT: DEGREE,
-        ATTR_UNIT_CONVERT: DEGREE,
-    },
-    "barometricPressure": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_PRESSURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Barometric Pressure",
-        ATTR_UNIT: PRESSURE_PA,
-        ATTR_UNIT_CONVERT: PRESSURE_INHG,
-    },
-    "seaLevelPressure": {
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_PRESSURE,
-        ATTR_ICON: None,
-        ATTR_LABEL: "Sea Level Pressure",
-        ATTR_UNIT: PRESSURE_PA,
-        ATTR_UNIT_CONVERT: PRESSURE_INHG,
-    },
-    "visibility": {
-        ATTR_DEVICE_CLASS: None,
-        ATTR_ICON: "mdi:eye",
-        ATTR_LABEL: "Visibility",
-        ATTR_UNIT: LENGTH_METERS,
-        ATTR_UNIT_CONVERT: LENGTH_MILES,
-    },
+
+class NWSSensorMetadata(NamedTuple):
```

Comment:
Any reason you prefer named tuple over data class ?

Reply:
`NamedTuple` are a bit more efficient compared to a `dataclass`. They are immutable and don't add the extra functionalities a `dataclass` has. Unless some of the values will change, I usually prefer a `NamedTuple`. It's easy enough to change it to a `dataclass` later :)

Some more details: https://github.com/home-assistant/core/pull/53233#discussion_r673269650